pkg/ingester/owned_series_test.go

```diff
@@ -1289,6 +1290,88 @@ func TestOwnedSeriesServiceWithPartitionsRing(t *testing.T) {
 	}
 }
 
+func TestOwnedSeriesStartsQuicklyWithEmptyIngesterRing(t *testing.T) {
+	ringKVStore, ringCloser := consul.NewInMemoryClient(ring.GetCodec(), log.NewNopLogger(), nil)
+	t.Cleanup(func() { assert.NoError(t, ringCloser.Close()) })
+
+	kvStore := &watchingKV{Client: ringKVStore}
+
+	rc := ring.Config{}
+	flagext.DefaultValues(&rc)
+
+	// Configure ring
+	rc.KVStore.Mock = kvStore
+	rc.HeartbeatTimeout = 1 * time.Minute
+	rc.ReplicationFactor = 1 // Number of zones must be equal to RF. We will only use 1 zone.
+	rc.ZoneAwarenessEnabled = true
+
+	rng := createAndStartRing(t, rc)
+
+	ringStrategy := newOwnedSeriesIngesterRingStrategy("ingester", rng, nil)
+
+	// Tenant checks are only done if ring is not empty.
+	tenantChecks := atomic.NewInt32(0)
+
+	const interval = 10 * time.Second
+	oss := newOwnedSeriesService(interval, ringStrategy, log.NewNopLogger(), nil, nil, func() []string { tenantChecks.Inc(); return []string{} }, nil)
+
+	// OwnedSeriesService will become Running immediately, but doesn't do any checks since ring is empty.
+	require.NoError(t, services.StartAndAwaitRunning(context.Background(), oss))
+	t.Cleanup(func() {
+		require.NoError(t, services.StopAndAwaitTerminated(context.Background(), oss))
+	})
+	require.Equal(t, int32(0), tenantChecks.Load())
+
+	// Add an instance to the ring. This is enough to start doing checks.
+	updateRingAndWaitForWatcherToReadUpdate(t, kvStore, func(desc *ring.Desc) {
+		desc.AddIngester("an-instance", "localhost:11111", "zone", []uint32{1, 2, 3}, ring.ACTIVE, time.Now())
+	})
+
+	// We should see owned series doing its checks now.
+	test.Poll(t, 2*(interval/10), true, func() interface{} {
+		return tenantChecks.Load() > 0
+	})
+}
+
+func TestOwnedSeriesWaitsForPartitionBeforeEnteringRunningState(t *testing.T) {
```

Comment:
nit:

Should this be `TestOwnedSeriesStartsQuicklyWithEmptyPartitionsRing`? We no longer wait for a partition before entering `Running`, right?